### PR TITLE
Update template.php

### PR DESCRIPTION
**Pull request by ricardoh34**
Include name of student and name of the course in the PDF certificate so as to ease organization of the certificates downloaded, solving issue #455

### CODE REVIEW

**Comment by mdjnelson** [comment]
@ricardoh34 are you able to address my peer review points?

**Comment by ricardoh34** [comment]
Hello Mark,

 I have seen that test fails, but I don´t understand why it happens, and
what I have to solve now. Can you give some advice?

Regards, Ricardo

**Comment by ricardoh34** [comment]
I see now the problem in SQL sentence. I have to change it to fullname, I
think

**Comment by ricardoh34** [comment]
I can´t see your peer review points, sorry, where can I see them? I saw the errors in the tests and I have changed the code to address properly fullname column in the database.

File: classes/template.php [hunk 1]
```diff
@@ -255,17 +255,15 @@ public function delete_element($elementid) {
...
 	if (empty($userid)) {
             $user = $USER;
         } else {
             $user = \core_user::get_user($userid);
         }
 
         require_once($CFG->libdir . '/pdflib.php');
-	
-        // Get the pages for the template, there should always be at least one page for each template.
+	// Get the pages for the template, there should always be at least one page for each template.
```
Comment (mdjnelson) [inline classes/template.php:266]: Why are you changing the spacing?

File: classes/template.php [hunk 2]
```diff
@@ -301,13 +299,12 @@ public function generate_pdf(bool $preview = false, int $userid = null, bool $re
                 $filename = get_string('certificate', 'customcert');
             }
             
-            $sql = "SELECT course,fullname as full_name
-                    FROM {customcert} c, 
-	            {course} cr
-                    WHERE c.id = :templateid
-			  AND c.course = cr.id";
+            $sql = "SELECT course,fullname
```
Comment (mdjnelson) [inline classes/template.php:302]: Please see https://docs.moodle.org/dev/SQL_coding_style#Indentation.

File: classes/template.php (hunk 1 above)
Comment (ricardoh34) [reply classes/template.php:266]: I didn´t realize, we can leave the original spacing

File: classes/template.php (hunk 2 above)
Comment (ricardoh34) [reply classes/template.php:302]: I thought I have corrected it already, I missed this

Comment (ricardoh34) [reply classes/template.php:303]: I think now it is right, "FROM {customcert} c, {course} cr"  is in one line and identation is correct. Hope now it is properly finished! Sorry for all the extra work

File: classes/template.php [hunk 3]
```diff
@@ -255,15 +255,16 @@ public function delete_element($elementid) {
      */
     public function generate_pdf(bool $preview = false, int $userid = null, bool $return = false) {
         global $CFG, $DB, $USER;
+	defined('MOODLE_INTERNAL') || die();
 
-        if (empty($userid)) {
+	if (empty($userid)) {
```
[the remark this answers is not in the document]
Comment (mdjnelson) [reply classes/template.php:259]: It seems you are using tabs, not spaces.

File: classes/template.php (hunk 1 above)
Comment (mdjnelson) [reply classes/template.php:267]: Please fix the spacing.

**Comment by mdjnelson** [comment]
Closing as there are still issues to address and I have a more efficient solution on solving this.

**Comment by mdjnelson** [comment]
I decided I would like to give credit to @ricardoh34. Can you please use spaces, instead of tabs and put this in one commit rebased on MOODLE_404_STABLE?

**Pull request closed without merging**